File: ReactiveCocoa/ReactiveCocoa/ReactiveCocoa/链式编程/ViewController.m
//
//  ViewController.m
//  ReactiveCocoa
//
//  Created by X-Liang on 2017/11/16.
//  Copyright © 2017年 X-Liang. All rights reserved.
//

#import "ViewController.h"
#import "CalculationMachine.h"
@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self testCalculationMachine];
}

- (void)testCalculationMachine {
    CalculationMachine *machine = [[CalculationMachine alloc] init];
//    [[[machine add:10] add:10] add:10];
    machine.addNum(10).addNum(10).addNum(10);
    NSLog(@"%ld", (long)machine.result);
}

@end

File: iOS/Swift/Portal-Swift/Portal-Swift/ViewController.swift
//
//  ViewController.swift
//  Portal-Swift
//
//  Created by Liang on 2018/4/23.
//  Copyright © 2018年 Liang. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
	override func viewDidLoad() {
		super.viewDidLoad()
	}
}

extension UIViewController {
	/// 当有很多页面跳转的时候，它会变得很凌乱。
	func openProfile(/*Other Parameters*/) {
		/** Use other Parameter to initial UIViewCotroller*/
		let viewController = UIViewController()
		present(viewController, animated: true, completion: nil)
	}
}

